salt/cloud.sls

```diff
@@ -32,49 +32,41 @@ salt-cloud:
       {% endif %}
 {% endif %}
 
-{% for folder in salt_settings.cloud.folders %}
-{{ folder }}:
-  file.directory:
-    - name: /etc/salt/{{ folder }}
-    - user: root
-    - group: root
-    - file_mode: 744
-    - dir_mode: 755
-    - makedirs: True
-{% endfor %}
-
 {% for cert in pillar.get('salt_cloud_certs', {}) %}
 {% for type in ['pem'] %}
 cloud-cert-{{ cert }}-pem:
   file.managed:
-    - name: /etc/salt/cloud.providers.d/key/{{ cert }}.pem
+    - name: /etc/salt/pki/cloud/{{ cert }}.pem
     - source: salt://salt/files/key
     - template: jinja
     - user: root
     - group: root
     - mode: 600
+    - makedirs: True
     - defaults:
         key: {{ cert }}
         type: {{ type }}
 {% endfor %}
 {% endfor %}
 
-{% for providers in salt_settings.cloud.providers %}
-salt-cloud-profiles-{{ providers }}:
-  file.managed:
-    - name: /etc/salt/cloud.profiles.d/{{ providers }}.conf
-    - template: jinja
-    - source: salt://salt/files/cloud.profiles.d/{{ providers }}.conf
-
-salt-cloud-providers-{{ providers }}:
-  file.managed:
-    - name: /etc/salt/cloud.providers.d/{{ providers }}.conf
+{%- for dir in ['providers', 'profiles', 'maps'] %}
+{%- set source = salt_settings.cloud.template_sources.get(dir) %}
+salt-cloud-{{ dir }}:
+  file.recurse:
+    - name: /etc/salt/cloud.{{ dir }}.d
```

Comment:
I believe the reason I used /etc/salt was because the rest of the state already used it. But I don't object to changing it.